pennylane/ops/op_math/decompositions/unitary_decompositions.py

```diff
@@ -464,9 +464,96 @@ def _extract_su2su2_prefactors(U, V):
     A, B = math.decomposition.su2su2_to_tensor_products(AB)
     C, D = math.decomposition.su2su2_to_tensor_products(CD)
 
```

Comment:
A main reason of the mentioned failed decomposition is that this `extraction` is not guaranteed to give a correct result. Also, we do not have a feedback system inside to help provide extra information and guidance whenever such errors actually happen. It took developers quite much effort to figure out where exactly went wrong. 
This drawback could be within a system-wide problem and we might need to start to think of how to combine the differentiability and error traceability.